src/plugins/walletConnectToDapps/components/DappRegistryGrid.tsx

```diff
@@ -0,0 +1,96 @@
+import { SearchIcon } from '@chakra-ui/icons'
+import {
+  Box,
+  Heading,
+  Image,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  Link,
+  SimpleGrid,
+  Stack,
+  Text as PlainText
+} from '@chakra-ui/react'
+import { Text } from 'components/Text'
+import { FC, useEffect, useMemo } from 'react'
+import { useForm } from 'react-hook-form'
+
+interface RegistryItem {
+  category: string
+  id: string
+  homepage: string
+  name: string
+  image: string
+}
+
+const registry: RegistryItem[] = require('../registry.json')
+
+const pageSize = 20
+
+export const DappRegistryGrid: FC = () => {
+  const { register, watch, setValue } = useForm<{ search: string; page: number }>({
+    mode: 'onChange',
+    defaultValues: {
+      search: '',
+      page: 0,
+    },
+  })
+
+  const search = watch('search')
+  const page = watch('page')
+  useEffect(() => setValue('page', 0), [search, setValue])
+
+  const filteredListings = useMemo(
+    () => registry.filter(l => !search || l.name.toLowerCase().includes(search.toLowerCase())),
+    [search],
+  )
+
+  return (
+    <Box>
+      <Stack direction='row' alignItems='center' mb={4}>
+        <Heading flex={1} fontSize='2xl'>
+          <Text translation='plugins.walletConnectToDapps.availableDapps' />
+        </Heading>
+        <Box>
+          <InputGroup>
+            <InputLeftElement pointerEvents='none'>
+              <SearchIcon color='gray.700' />
+            </InputLeftElement>
+            <Input
+              {...register('search')}
+              type='text'
+              placeholder='Search'
+              pl={10}
+              variant='filled'
+            />
+          </InputGroup>
+        </Box>
+      </Stack>
+      <SimpleGrid columns={{ lg: 4, sm: 2, base: 1 }} spacing={4}>
+        {filteredListings.slice(page * pageSize, pageSize).map(listing => (
```

Comment:
`slice` creates a new ref every single time - can we please rip this out and `useMemo` this part of the tree?

Reply:
Does that matter? This is just mapping over a certain set of elements (`listing`) which are objects that don't change, regardless of if they're sliced. Aka `const array = [{}]; array[0] === array.slice()[0] ===> true`. More than happy to memoize though, I'm just not sure it matters performance-wise